sktime/transformers/series_as_features/signature_based/_window.py

```diff
@@ -0,0 +1,189 @@
+"""
+window.py
+======================================
+Introduces the Window module that is used when splitting the path over:
+    - Global
+    - Sliding
+    - Expanding
+    - Dyadic
+window types.
+
+Code based on window code written by Patrick Kidger.
+"""
+import collections as co
+
+
+_Pair = co.namedtuple('Pair', ('start', 'end'))
+
+
+def window_getter(window_name,
+                  window_depth=None,
+                  window_length=None,
+                  window_step=None
+                  ):
+    """Gets the window method correspondent to the given string and initialises
+    with specified parameters.
+
+    Parameters
+    ----------
+    window_name: str, String from ['global', 'sliding', 'expanding', 'dyadic']
+        used to access the window method.
+    window_depth: int, The depth of the dyadic window. (Active only if
+        `window_name == 'dyadic']`.
+    window_length: int, The length of the sliding/expanding window. (Active
+        only if `window_name in ['sliding, 'expanding'].
+    window_step: int, The step of the sliding/expanding window. (Active
+        only if `window_name in ['sliding, 'expanding'].
+
+    Returns
+    -------
+    list:
+        A list of lists where the inner lists are lists of tuples that
+        denote the start and end indexes of each window.
+    """
+    # Setup all available windows here
+    length_step = {'length': window_length, 'step': window_step}
+    window_dict = {
+        'global': (Global, {}),
+        'sliding': (Sliding, length_step),
+        'expanding': (Expanding, length_step),
+        'dyadic': (Dyadic, {'depth': window_depth}),
+    }
+
+    assert window_name in window_dict.keys(), (
+        "Window name must be one of: {}. Got: {}."
+        "".format(window_dict.keys(), window_name)
+    )
+
+    window_cls, window_kwargs = window_dict[window_name]
+    return window_cls(**window_kwargs)
+
+
+class Window:
+    """Abstract base class for windows.
+
+    Each subclass must implement a __call__ method that returns a list of lists
+    of 2-tuples. Each 2-tuple specifies the start and end of each window.
+    These windows are grouped into a list that will (usually) cover the full
+    time series. These lists are grouped into another list for situations
+    where we consider windows of multiple scales.
+    """
+    def num_windows(self, length):
+        """Method that returns the total number of windows in the set.
+
+        Parameters
+        ----------
+        length: int, The length of the input path.
+
+        Returns
+        -------
+        int: The number of windows.
+        """
+        return sum([len(w) for w in self(length)])
+
+
+class Global(Window):
```

Comment:
same here, I'd suggest making this a private class, unless it's meant to be used directly by users, otherwise perhaps a somewhat more descriptive name